src/EventListener/SerializeListener.php

```diff
@@ -61,7 +61,9 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
 
         $context = $this->serializerContextBuilder->createFromRequest($request, true, $attributes);
 
-        if (isset($context['output']) && \array_key_exists('class', $context['output']) && null === $context['output']['class']) {
+        if ((isset($context['output']) && \array_key_exists('class', $context['output']) && null === $context['output']['class']) ||
+        (null === $controllerResult && isset($context['input']) && \array_key_exists('class', $context['input']) && null === $context['input']['class'])
+        ) {
```

Comment:
this is if is sort of unreadable now.
can it be improved?

Reply:
I've done my best see updated code :p.